### PR TITLE
Test with Local Delivery and rename docker config to kitchen.dokken.yml

### PR DESCRIPTION
**Pull request by tas50**
Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by tas50** [comment]
This fixes some Chef 13 issues too and turns on the deprecation warnings

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -8,24 +8,9 @@ transport:
 
 provisioner:
   name: dokken
+  deprecations_as_errors: true
```
Comment (damacus) [inline .kitchen.dokken.yml]: Doesn't this get inherited as part of .kitchen.yml?

File: test/cookbooks/test/recipes/ruby.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 file '/tmp/pg_gem_test.rb' do
   content "#!/opt/chef/embedded/bin/ruby\nrequire 'pg'"
-  mode 0755
+  mode '755'
```
Comment (damacus) [inline test/cookbooks/test/recipes/ruby.rb]: Why? :/

Comment (tas50) [inline test/cookbooks/test/recipes/ruby.rb]: Using strings here allow you to preserve any leading 0s. It's a rubocop-chef rule actually, which is what fixed this.

File: .kitchen.dokken.yml (hunk 1 above)
Comment (tas50) [inline .kitchen.dokken.yml]: That's a great question. I'll have to confirm that

**Pull request merged**